### PR TITLE
Unify FilesList and SharesList

### LINKED ISSUE #1218: Unify ShareList and FileList
As per https://github.com/ChainSafe/files-ui/pull/1048#pullrequestreview-695997426 those 2 are very similar and we should ultimately use only 1.

### PR DESCRIPTION
**Pull request by Tbaut**
closes #1218

### CODE REVIEW

**Review by FSM1 — APPROVED** [review]

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx [hunk 1]
```diff
@@ -301,7 +304,8 @@ const FilesList = () => {
     moduleRootPath,
     isSearch,
     withSurvey,
-    bucket
+    bucket,
+    accessRole
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx]: Don't we have the `accessRole` inside `bucket` (BucketKeyPermission) now ?

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx [hunk 2]
```diff
@@ -466,17 +470,52 @@ const FilesList = () => {
...
+      "preview",
+      "rename"
+    ]
+
+    if (!!accessRole && isShared) {
+
+      switch(accessRole) {
+      case "owner":
+        filteredList = [
+          "delete",
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx]: could we have something like this

`const readerPermissions = ["info", "download"]`
`const writerPermissions = [...readerPermissions, "move", ....]`

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx [hunk 3]
```diff
@@ -466,17 +470,52 @@ const FilesList = () => {
...
-        "info",
-        "move",
-        "preview",
-        "rename",
-        "share",
-        "recover"
-      ]
+    if (!bulkOperations) return
+
+    let filteredList: FileOperation[] = [
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx]: maybe `fileOperations` a better name ?

**Review by tanmoyAtb — COMMENTED** [review]
Working great! 💯 
Left somee comments

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx (hunk 1 above)
Comment (Tbaut) [reply packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx]: Totally, I merged dev and removed this prop

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx (hunk 2 above)
Comment (Tbaut) [reply packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx]: Love the idea, check 952b03e and let me know if that's better.

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx (hunk 3 above)
Comment (Tbaut) [reply packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx]: agreed, changed.

**Review by tanmoyAtb — APPROVED** [review]
Looking great ! 👍

**Pull request merged**